cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -670,30 +669,30 @@ fromShelleyTxOut era ledgerTxOut =
 
     ShelleyBasedEraBabbage -> error "TODO: Babbage"
 
-toAlonzoTxOutDataHash :: TxOutDatum CtxUTxO era
-                      -> StrictMaybe (Alonzo.DataHash StandardCrypto)
+-- TODO: If ledger creates an open type family for datums
+-- we can consolidate this function with the Babbage version
+toAlonzoTxOutDataHash
+  :: TxOutDatum CtxUTxO AlonzoEra
+  -> StrictMaybe (Alonzo.DataHash StandardCrypto)
 toAlonzoTxOutDataHash  TxOutDatumNone                        = SNothing
 toAlonzoTxOutDataHash (TxOutDatumHash _ (ScriptDataHash dh)) = SJust dh
-toAlonzoTxOutDataHash (TxOutInlineDatum _ sd) =
-  let ScriptDataHash alonzoDataHash = hashScriptData sd
-  in SJust alonzoDataHash
+toAlonzoTxOutDataHash (TxOutDatumInline inlineDatumSupp _sd) =
+  case inlineDatumSupp :: InlineDatumSupportedInEra AlonzoEra of {}
 
 fromAlonzoTxOutDataHash :: ScriptDataSupportedInEra era
                         -> StrictMaybe (Alonzo.DataHash StandardCrypto)
                         -> TxOutDatum ctx era
 fromAlonzoTxOutDataHash _    SNothing  = TxOutDatumNone
-fromAlonzoTxOutDataHash era (SJust dh) = TxOutDatumHash era (ScriptDataHash dh)
-
-_toBabbageTxOutDatum
-  :: Ledger.Crypto (ShelleyBasedEra era) ~ StandardCrypto
-  => TxOutDatum CtxUTxO era -> Babbage.Datum (ShelleyBasedEra era)
-_toBabbageTxOutDatum  TxOutDatumNone                        = Babbage.NoDatum
-_toBabbageTxOutDatum (TxOutDatumHash _ (ScriptDataHash dh)) = Babbage.DatumHash dh
-_toBabbageTxOutDatum (TxOutInlineDatum _ sd) = scriptDataToBinaryData sd
--- TODO: Rename TxOutDatum' to TxOutAddDatumToDatumMap or something similar
--- as we need to distinguish between this feature (which is an Alonzo feature)
--- and the inline datum feature (which is a Babbage feature).
--- toBabbageTxOutDatum (TxOutDatum' _ (ScriptDataHash dh) _) = Babbage.DatumHash dh
+fromAlonzoTxOutDataHash s (SJust dh)   = TxOutDatumHash s (ScriptDataHash dh)
+
+-- TODO: If ledger creates an open type family for datums
+-- we can consolidate this function with the Alonzo version
+toBabbageTxOutDatum
+  :: Ledger.Crypto (ShelleyLedgerEra era) ~ StandardCrypto
+  => TxOutDatum CtxUTxO era -> Babbage.Datum (ShelleyLedgerEra era)
+toBabbageTxOutDatum  TxOutDatumNone = Babbage.NoDatum
+toBabbageTxOutDatum (TxOutDatumHash _ (ScriptDataHash dh)) = Babbage.DatumHash dh
+toBabbageTxOutDatum (TxOutDatumInline _ sd) = scriptDataToBinaryData sd
```

Comment:
The code looks the same, but the TODO is gone.  Is the TODO not needed anymore?

Reply:
It's not needed anymore as `TxOutDatum'` was renamed to `TxOutDatumInTx'`